### DISCUSSION TITLE
Unable to connect to remote WebDAV server

### DISCUSSION THREAD

**Discussion by sirkelsalot** [photoprism/photoprism#2617] [Q&A]
#### 1. What is not working as documented?

Trying to connect from Photoprism docker instance on Unraid to a SeaFile WebDAV docker instance on the same server, but I'm unable to connect. At this time, SeaFile WebDAV only supports using an email address as the WebDAV username.

I followed the instructions on the FAQ to use CURL within the Photoprism docker container to verify connectivity. The curl command works successfully from within the Photoprism container.
`curl -X PROPFIND -H "Depth: 1" -u email@address.com:long-pass-phrase https://dav.domain.com/My%20Photos/`

However, entering the same info in the Photoprism UI returns "Could not connect, please try again". I've tried the direct IP:port without going through the reverse proxy as well, but same result. Both work fine through the suggested CURL command, but not Photoprism UI.

https://docs.photoprism.app/user-guide/faq/#webdav

#### 2. How can we reproduce it?

1. Settings, Sync, Add Server
2. Enter target WebDAV path (Seafile for me)
3. Enter an email address as the user id for the target WebDAV server and appropriate pass phrase
4. Click Connect, receive error

**Comment by StephanGeorg** [comment]
Same here. 
Log output: `ERRO could not connect`

**Reply by lastzero** [reply]
What specifically? Can you provide any details like version, URL,...?

**Comment by eculicny** [comment]
I seem to be encountering a similar problem while trying to connect to a TrueNAS WebDAV share.

Environment: Docker
Photoprism version: `230719-73fa7bbe8-Linux-AMD64-Plus`
(edit: Docker host is physically separate from the TrusNAS host)

Recommended curl command works as expected and prints the XML for the directory and the items within it.
```curl -X PROPFIND -H "Depth: 1" -u <user>:<pwd> http://<ip>:<port>/photosimport/```

When setting up this connection in the UI, all that pops up is the `Could not connect, please try again` message at the bottom. The network request to the Photoprism API is also returning 400 `Could not connect, please try again`


The only thing I can think of being an issue is if Photoprism enforces use of `https`, but didn't find any restrictions (just validating non-expired certs for `https`).

Is there anything else specifically I should look into here?

Thanks!

**Reply by lastzero** [reply]
Does that curl command work from within the PhotoPrism Docker container? If not, it could be a networking issue. Another thing to investigate is the custom port as I've never actively tested this, so I can't say with confidence that this might not cause problems (although it should not, of course).

**Reply by eculicny** [reply]
Forgot to specify that piece of it. The command does work as expected from within the docker container, so probably not a networking issue.

Thanks! I'll take a look and see if something shows up in debug logs or network requests.

**Reply by eculicny** [reply]
Can confirm that it's definitely the port causing the problem with adding/saving the service.

When using the url with port, the failure is due to a 400 error from `api/v1/services`:
```time="2023-07-27T00:48:41Z" level=debug msg="api-v1: abort /api/v1/services with code 400 (could not connect, please try again)"```

If I remove the port, the service is added successfully, but the connection doesn't work for the obvious reason that port 80 isn't the correct one.

I took a quick look at the code for that endpoint. Unfortunately, I'm not familiar enough with Go/Gin to determine where the problem exactly is without spinning up a dev environment, but best guess is something in the [discover](https://github.com/photoprism/photoprism/blob/develop/internal/remote/discover.go#L20) logic due to the generic "could not connect" message.

**Reply by lastzero** [reply]
Thanks for testing! In that case, I assume there's a problem with the URL parser. You're welcome to create a bug issue (including examples of URLs that don't work) so we can fix it for you.
